Vagrantfile

```diff
@@ -150,6 +150,64 @@ def packages_netbsd
   EOF
 end
 
+def packages_cygwin
+  return <<-EOF
+    mkdir -p /cygdrive/c/cygwin
+    powershell -Command '$client = new-object System.Net.WebClient; $client.DownloadFile("https://www.cygwin.com/setup-x86.exe","C:\\cygwin\\setup-x86_64.exe")'
+    echo '
+    REM --- Change to use different CygWin platform and final install path
+    set CYGSETUP=setup-x86_64.exe
+    REM --- Install build version of CygWin in a subfolder
+    set OURPATH=%cd%
+	    set CYGBUILD=%OURPATH%\\CygWin
+	    set CYGMIRROR=ftp://mirrors.kernel.org/sourceware/cygwin/
+	    set BUILDPKGS=python3,python3-setuptools,binutils,gcc-g++,libopenssl,openssl-devel,git,make,openssh,liblz4-devel,liblz4_1,rsync,curl,python-devel
+    %CYGSETUP% -q -B -o -n -R %CYGBUILD% -L -D -s %CYGMIRROR% -P %BUILDPKGS%
+	    ' > /cygdrive/c/cygwin/install.bat
+    cd /cygdrive/c/cygwin && cmd.exe /c install.bat
+    echo "alias mkdir='mkdir -p'" > ~/.profile
+    echo "export CYGWIN_ROOT=/cygdrive/c/cygwin/CygWin/bin" >> ~/.profile
+    echo 'export PATH=$PATH:$CYGWIN_ROOT' >> ~/.profile
+    cmd.exe /c 'setx /m PATH "%PATH%;C:\\cygwin\\CygWin\\bin"'
+    source ~/.profile
+  EOF
+end
+
+def build_cygwin_venv()
+  return <<-EOF
+    /cygdrive/c/cygwin/CygWin/bin/bash.exe -c "
+      easy_install-3.4 pip
+      pip install virtualenv
+      cd $HOME
+      virtualenv --python=python3 borg-env
+    "
+  EOF
+end
+
+def cygwin_install_borg(python_version)
+  return <<-EOF
+    /cygdrive/c/cygwin/CygWin/bin/bash.exe -c "
+      cd $HOME
+      . borg-env/bin/activate
+      cd borg
+      rm -f borg/*.so borg/*.cpy*
+      rm -f borg/{chunker,crypto,compress,hashindex,platform_linux}.c
+      rm -rf borg/__pycache__ borg/support/__pycache__ borg/testsuite/__pycache__
+      pip install -r requirements.d/development.txt
+      pip install -e .
+    "
```

Comment:
``sh`` vs ``bash`` mess is caused by mls-openssh ssh server installed there, here it is (related)[https://github.com/joefitzgerald/packer-windows/issues/123] discussion in packer-windows project that provides some details. I'll try to find a way to replace that ssh deamon with a cygwin one, otherwise I'll change default shell client on first run